libass/ass_render.c

```diff
@@ -978,7 +978,20 @@ ASS_Vector ass_layout_res(ASS_Renderer *render_priv)
         return (ASS_Vector) { settings->storage_width, settings->storage_height };
 
     ASS_Track *track = render_priv->track;
-    return (ASS_Vector) { track->PlayResX, track->PlayResY };
+    if (settings->par <= 0 || settings->par == 1)
+        return (ASS_Vector) { track->PlayResX, track->PlayResY };
+    if (settings->par > 1)
+        return (ASS_Vector) {
+            lround(track->PlayResY * render_priv->orig_width / render_priv->orig_height
```

Comment:
This could benefit from an `orig_height != 0` guard (and similarly below). We do have such a guard in the default `par` calculation in `ass_start_frame`.

Reply:
Good catch, thanks, now fixed!
I guess zero area `frame_content_*` is allowed, to avoid margin-using dialogue lines disappearing if the video is zoomed to zero?

Reply:
TBVH I can’t say why or even _whether_ it is “allowed”, and I’d rather it wasn’t, given that it completely breaks this and the par calculation. But I guess it ultimately boils down to: what other option do we have, if someone does give us a zero? Just refuse to render anything at all?